integration/library_test.go

```diff
@@ -105,14 +113,55 @@ var testCases = []testCase{
 	},
 }
 
+func TestFanal_Library_DockerLessMode(t *testing.T) {
+	for _, tc := range testCases {
+		tc := tc // save a copy of tc for use in t.Run https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			ctx := context.Background()
+			d, _ := ioutil.TempDir("", "TestFanal_Library_DockerLessMode_*")
+			defer os.RemoveAll(d)
+			c := cache.Initialize(d)
+
+			opt := types.DockerOption{
+				Timeout:  600 * time.Second,
+				SkipPing: true,
+			}
+
+			cli, err := client.NewClientWithOpts(client.FromEnv)
+			require.NoError(t, err, tc.name)
+
+			// clear existing Cache if any
+			require.NoError(t, c.Clear(), tc.name)
+
+			// remove existing Image if any
+			_, _ = cli.ImageRemove(ctx, tc.dockerlessImageName, dtypes.ImageRemoveOptions{
+				Force:         true,
+				PruneChildren: true,
+			})
+
+			ext, err := docker.NewDockerExtractor(opt, c)
+			require.NoError(t, err, tc.name)
+			ac := analyzer.Config{Extractor: ext}
+
+			// run tests twice, one without cache and with cache
+			for i := 1; i <= 2; i++ {
+				runChecks(t, "dockerless", ac, ctx, tc, d, c, nil)
+			}
+
+			// clear Cache
+			require.NoError(t, c.Clear(), tc.name)
+		})
+	}
+}
+
 func TestFanal_Library_DockerMode(t *testing.T) {
-	t.Parallel()
```

Comment:
I have removed these in favour of stability. I saw a race condition (didn't dive into why it was happening) amongst Tar and DockerMode tests when run in parallel (similar to what we saw previously). Also adding parallelism at this level doesn't save us much, in my tests it was only 1-2 seconds faster with `t.Parallel`